### PR TITLE
More 0.6.1 updates

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/spec/beaconstate.nim [hunk 1]
```diff
@@ -120,27 +120,43 @@ func get_delayed_activation_exit_epoch*(epoch: Epoch): Epoch =
...
+
+  # Return if validator already initiated exit
+  let validator = addr state.validator_registry[index]
+  if validator.exit_epoch != FAR_FUTURE_EPOCH:
+    return
+
+  # Compute exit queue epoch
+  let exit_epochs = mapIt(
+    filterIt(state.validator_registry, it.exit_epoch != FAR_FUTURE_EPOCH),
+    it.exit_epoch)
```
Comment (mratsim) [inline beacon_chain/spec/beaconstate.nim]: Offtopic: probably the kind of place where we want to introduce zero-functional

File: beacon_chain/spec/beaconstate.nim [hunk 2]
```diff
@@ -284,9 +291,10 @@ func get_genesis_beacon_state*(
 
   # Process genesis activations
   for validator_index in 0 ..< state.validator_registry.len:
-    let vi = validator_index.ValidatorIndex
-    if get_effective_balance(state, vi) >= MAX_EFFECTIVE_BALANCE:
-      activate_validator(state, vi, true)
+    var validator = addr state.validator_registry[validator_index]
```
Comment (mratsim) [inline beacon_chain/spec/beaconstate.nim]: Should be `let`.

Alternatively for aliasing without pointers we can use

```Nim
template validator(): untyped {.dirty.} = state.validator_registry[validator_index]
```

Maybe we should add an "alias" template in helpers like so:

```Nim
template alias(ident, expression: expression): untyped =
  template ident(): untyped = expression
```

And used:

```Nim
alias validator, state.validator_registry[validator_index]
```

I expect the field accesses and repeated indexing to be optimized away by the compiler.

File: beacon_chain/spec/beaconstate.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/spec/beaconstate.nim]: Yes, I absolutely agree. This is a good example of where I'm prepared to be minorly wasteful of memory allocation, GC time, etc unless profiling shows it as as hotspot, but zero-functional would work ideally in this sort of straightforward case. Some of the examples they show are nearly exactly this, its potential limits regarding first-class reification of composed filters, say, notwithstanding.

File: beacon_chain/spec/beaconstate.nim (hunk 2 above)
Comment (tersec) [reply beacon_chain/spec/beaconstate.nim]: Addressed with https://github.com/status-im/nim-beacon-chain/pull/274/commits/21ef3d98215a20ddf6cce1bf3667bbea7a3b2f1a

**Pull request merged**